source/as-promise/index.ts

```diff
@@ -20,6 +20,26 @@ const proxiedRequestEvents = [
 	'downloadProgress'
 ];
 
+// Thanks https://github.com/drpicox/async-barrier
+/* eslint-disable no-undef-init */
+function makeAsyncBarrier() {
```

Comment:
There's no need to have this function...

Reply:
Look at the last commit, I tried to make something reusable (maybe it'll be needed in the future), also is cleaner than before.